### PR TITLE
ENH: Bump SWIG to 2024-03-26-master

### PR DESCRIPTION
**Pull request by thewtex**
- Supports more modern C++
- More Pythonic interfaces, e.g. Python Iterator Protocol
- Python Limited ABI

There were duplicate definition errors due to double inclusion of

  %template(vectoritkImageXX, std::vector< itkImageXX >)

via itkImage_ext.i in PCAShapeSignedDistanceFunction. Remove
itkImage_ext.i from the latter but still provide the required typemaps
by adding DECL_PYTHON_STD_VEC_RAW_TO_SMARTPTR_TYPEMAP in
itkPCAShapeSignedDistanceFunction.wrap.

Remove -py3 because it is no longer supported (the default).

Remove "ios_base*" config templates because they are no longer emitted.

Bump PCRE. SWIG and PCRE are now built via CMake instead of autotools.

Closes https://github.com/InsightSoftwareConsortium/ITK/issues/4536
Closes https://github.com/InsightSoftwareConsortium/ITK/issues/3364
Closes https://github.com/InsightSoftwareConsortium/ITK/issues/3955

### CODE REVIEW

**Comment by thewtex** [comment]
@dzenanz note how to support `std::vector< ImageType >` on class APIs.

**Comment by thewtex** [comment]
Closes #4536

**Comment by N-Dekker** [comment]
Wow, that's a big step forward! Thanks Matt!

Still wondering, is there a specific reason to take the revision of 2024-03-26-master, rather than the last official release, https://github.com/swig/swig/releases/tag/v4.2.1  ?

**Comment by thewtex** [comment]
> Still wondering, is there a specific reason to take the revision of 2024-03-26-master, rather than the last official release, https://github.com/swig/swig/releases/tag/v4.2.1 ?

Yes, we want this patch: https://github.com/swig/swig/commit/21aa6e8f9e4c7d1b0249c46140b7ce5397d7dc52

**Comment by dzenanz** [comment]
Does this touch #3955?

File: Modules/Segmentation/SignedDistanceFunction/wrapping/itkPCAShapeSignedDistanceFunction.wrap [hunk 1]
```diff
@@ -8,6 +8,7 @@ foreach(d ${ITK_WRAP_IMAGE_DIMS})
   # `double` because of OptimizerParameters is hardcoded as templated
   # over `double` in parent class `ShapeSignedDistanceFunction`.
   foreach(t ${types})
+    string(APPEND ITK_WRAP_PYTHON_SWIG_EXT "DECL_PYTHON_STD_VEC_RAW_TO_SMARTPTR_TYPEMAP(itkImage${ITKM_${t}}${d}, itkImage${ITKM_${t}}${d}_Pointer)\n")
```
Comment (dzenanz) [inline Modules/Segmentation/SignedDistanceFunction/wrapping/itkPCAShapeSignedDistanceFunction.wrap:11]: Is this expected to work in remote modules? Should I replace [the old way](https://github.com/dzenanz/ANTsWasm/commit/f7e72cf5bb2a11837cb18aa13166cd2a56652221) with this style?

**Comment by dzenanz** [comment]
MacOS Python build error:
```log
CMake Error at /usr/local/Cellar/cmake/3.28.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find BISON: Found unsuitable version "2.3", but required is at
  least "3.5" (found /usr/bin/bison)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.28.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:598 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.28.1/share/cmake/Modules/FindBISON.cmake:306 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:91 (find_package)
```

**Review by dzenanz — APPROVED** [review]
Code looks good. Maybe we need a newer MacOS virtual machine image?

**Comment by dzenanz** [comment]
I assume we need to bump the version here:
https://github.com/InsightSoftwareConsortium/ITK/blob/fad58494684c15c71525e127ba5fc6318dd33831/Testing/ContinuousIntegration/AzurePipelinesMacOS.yml#L34

**Comment by thewtex** [comment]
> Does this touch https://github.com/InsightSoftwareConsortium/ITK/issues/3955?

I will add macOS binaries.

File: Modules/Segmentation/SignedDistanceFunction/wrapping/itkPCAShapeSignedDistanceFunction.wrap (hunk 1 above)
Comment (thewtex) [reply Modules/Segmentation/SignedDistanceFunction/wrapping/itkPCAShapeSignedDistanceFunction.wrap:11]: Yes

**Comment by thewtex** [comment]
> I assume we need to bump the version here:

No, we want to keep testing against the older version, and bumping would not help with the build.

**Review by dzenanz — APPROVED** [review]

**Pull request merged**